sql/migrate/migrate.go

```diff
@@ -171,6 +188,29 @@ type (
 
 	// PlannerOption allows managing a Planner using functional arguments.
 	PlannerOption func(*Planner)
+
+	// A Migration denotes an applied migration in a deployment. Used to track migration executions state of a database.
+	Migration struct {
+		// Version of the migration.
+		Version string
+		// Description of this migration.
+		Description string
+		// ExecutionState if this migration. One of ["ongoing", "ok", "error"].
+		ExecutionState string
+		// ExecutedAt denotes when this migration was started to be executed.
+		ExecutedAt time.Time
+		// ExecutionTime denotes the time it took for this migration to be applied on the database.
+		ExecutionTime time.Duration
+		// Hash is the check-sum of this migration as stated by the migration directories HashFile.
+		Hash string
+		// OperatorVersion holds a string representation of the Atlas operator managing this database migration.
+		OperatorVersion string
+		// Meta holds additional custom meta-data given for this migration.
+		Meta map[string]string
+	}
+
+	// State is an ordered set of Migration structs.
+	State []Migration
```

Comment:
3 comments about the naming here. It feels better for me to name the `migrate.Migration` struct above as `migrate.Revision`, the `State` below as something like `History` or `Log`, and the container above as follows:

```go
type HistoryReadWriter interface {
	ReadHisotry() (History, error)
	WriteRevision(Revision) error
}
```

Reply:
Rethinking about this, the `History` (revision list) feels redundant. You can also write the above interface as follows:

```go
type RevisionReadWriter interface {
	ReadRevisions() ([]Revision, error)
	WriteRevision(Revision) error
}
```

Reply:
I like `Revision` :-)